dashboard/frontend/src/pages/Job.tsx

```diff
@@ -180,8 +223,10 @@ const JobPage: FC<{
   const closeExecModeration = useCallback(async () => { setExecModWindowOpen(false) }, [])
 
   useEffect(() => {
-    loadInfo()
-  }, [])
+    loadInfo();
+    loadInputRelationInfo();
+    loadOutputRelationInfo();
+  }, [loadInfo, loadInputRelationInfo]);
```

Comment:
With `useEffect` - the array that you pass is slightly different that with `useCallback` or `useMemo`

In all cases - it's "if any of these change then redo"

In the case of `useCallback` or `useMemo` - that means "re-memoize" - i.e. incorporate the latest (changed) values in the function or value

In the case of `useEffect` - it means `re-run this function` - I think the intended use-case here is:

> when the id changes then reload the stuff from the API

If that is true then it makes more sense to have:


```ts
useEffect(() => {
    loadInfo();
    loadInputRelationInfo();
    loadOutputRelationInfo();
  }, [id]);
```

I.e. "when id changes - re-run these 3 functions"

ack that this is how it was before you made any changes and I've only just caught this now in this review :-)